README.md
# Linux
#### Linux is open source operating system
#### It is fast and secure
## Run a shell script
shell-name filename.sh
```bash 
bash wifi_driver.sh
```


history/history.txt
    1  passwd root
    2  cd
    3  exit
    4  yum install python3-pip -y
    5  pip --version
    6  pip
    7  pip3 --version
    8  pip3 install jupyter
    9  jupyter-notebook
   10  jupyter
   11  jupyter --version
   12  pip3 install jupyter
   13  jupyter notebook
   14  ls
   15  jupyter notebook
   16   ~/.local/bin/jupyter-notebook
   17  .local/bin/jupyter-notebook
   18  ./.local/bin/jupyter-notebook
   19  pip3 uninstall jupyter
   20  pip3 install jupyter
   21  which jupyter
   22  pip3 install jupyter-notebook
   23  pip3 install notebook
   24  python3 -m notebook
   25  cd ..
   26  exit
   27  passwd root
   28  useradd harsh
   29  su harsh
   30  passwd harsh
   31  su harsh
   32  dnf repolist
   33  dnf repolist -v
   34  dnf repolist -v | grep docker
   35  dnf remove docker
   36  dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo
   37  dnf list docker-ce
   38  dnf install docker-ce
   39  dnf config-manager \ --add-repo=https://download.docker.com/linux/centos/docker-ce.repo
   40  dnf install docker-ce docker-ce-cli 
   41  dnf install docker-ce docker-ce-cli containerd.io
   42  dnf install docker-ce docker-ce-cli containerd.io --nobest
   43  yum list
   44  yum list | grep docker
   45  yum autoremove
   46  sudo yum install docker-ce-3:18.09.1-3.el7
   47  systemctl start docker
   48  docker images
   49  sudo yum install docker-ce-cli-1:19.03.8-3.el7  
   50  docker run hello-world
   51  docker images
   52  docker ps
   53  docker ps -a
   54  docker pull python
   55  docker images
   56  docker run python
   57  docker ps 
   58  docker ps python
   59  docker ps python3
   60  docker ps -a
   61  pwd
   62  cat >hello.py
   63  cat hello.py 
   64  python3  hello.py 
   65  ls
   66  vim hello.py 
   67  dnf instll vim -y
   68  yum install vim -y
   69  vim hello.py 
   70  docker run python hello.py 
   71  python3 hello.py 
   72  vim hello.py 
   73  python3 hello.py 
   74  vim hello.py 
   75  python3 hello.py 
   76  vim hello.py 
   77  python3 hello.py 
   78  vim hello.py 
   79  python3 hello.py 
   80  vim DockerFile
   81  pwd
   82  shutdown
   83  shutdown --help
   84  shutdown -P
   85  poweoff
   86  poweroff
   87  su -i
   88  su harsh
   89  yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
   90  cd /etc/yum.repos.d/
   91  ls
   92  cd
   93  dnf
   94  dnf install docker
   95  docker
   96  docker --version
   97  ls
   98  export DOCKER_HOST="tcp://52.204.127.145:2375"
   99  docker version
  100  docker info
  101  docker run hello.py 
  102  docker images
  103  docker run ubuntu
  104  docker ps
  105  docker stop ubuntu
  106  docker stop 9a6182db9278
  107  docker search mongodb
  108  docker images
  109  history
  110  poweroff
  111  git clone https://github.com/harshmangal1108/dockerapp/tree/master
  112  dnf install git
  113  git clone https://github.com/harshmangal1108/dockerapp/tree/master
  114  git clone https://github.com/harshmangal1108/dockerapp
  115  ls
  116  docker bulid -t harshpy:v1 https://github.com/harshmangal1108/dockerapp.git
  117  history 
  118  export DOCKER_HOST="tcp://52.204.127.145:2375"
  119  docker info
  120  ls
  121  useradd dateuser
  122  passwd dateuser
  123  sudo chmod a-x /usr/bin/date 
  124  date
  125  setfacl -m u:dateuser:x /usr/bin/date
  126  su dateuser
  127  exit
  128  setfacl -m u:dateuser:000 /*
  129  setfacl -m u:dateuser:000 /
  130  su dateuser
  131  setfacl -m u:dateuser:r /
  132  su dateuser
  133  setfacl -m u:dateuser:rwx /bin/bash
  134  su dateuser
  135  setfacl -m u:dateuser:rwx /
  136  su dateuser
  137  setfacl -m u:dateuser:rwx /
  138  setfacl -m u:dateuser:rwx /*
  139  setfacl -m u:dateuser:rwx /
  140  getfacl 
  141  userdel A B dateuser
  142  userdel -r A
  143  userdel -r B
  144  userdel -r dateuser
  145  cd /home/
  146  sl
  147  ls
  148  rm -rvf dateuser/
  149  docker run hello-world
  150  docker run -it --name myct1 alpine sh
  151  docker rm myct1
  152  docker run -it --name myct1 alpine sh
  153  docker volume create hvol1
  154  docker volume ls
  155  docker run -it --name myct1 -v hvol1:/mydata   alpine sh
  156  docker run -it --name myct2 -v hvol1:/mydata   alpine sh
  157  docker rm myct2
  158  docker run -it --name myct2 -v hvol1:/mydata   alpine sh
  159  docker run -it --name myct3 -v hvol1:/harsh  java bash
  160  history 
  161  docker images
  162  cd /var/lib/docker
  163  ls
  164  cd volumes/
  165  ls
  166  cd hvol1/
  167  ls
  168  cd _data
  169  ls
  170  cat abc 
  171  mkdir abcd
  172  ls
  173  cd
  174  docker run -it --name myct3 -v hvol1:/harsh  java bash
  175  docker run -it --name myct4 -v hvol1:/harsh  java bash
  176  docker ps 
  177  docker images
  178  history 
  179  poweroff 
  180   DOCKER_HOST="tcp://52.204.127.145:2375" 
  181  docker info
  182  DOCKER_HOST="tcp://52.204.127.145:2375" 
  183  DOCKER_HOST="tcp://52.204.127.145:2375"
  184  docker info
  185  exit
  186  export DOCKER_HOST="tcp://52.204.127.145:2375"
  187  docker info'
  188  /
  189  \
  190  docker info
  191  docker images
  192  docker build -t apline:harshmangal
  193  docker run --it --name apline:harshmangalc1q1
  194  docker run -it --name apline:harshc1q1
  195  docker run -it --name apline:harshc1q1 alpine
  196  question1
  197  mkdir question1
  198  cd question1/
  199  ls
  200  vim Dockerfile
  201  docker build -t apline:harsh .
  202  docker images
  203  docker run -itd --name harshc1q1apline:harsh
  204  docker run -it --name harshc1q1a-d :harsh
  205  docker build -t alpine:harsh .
  206  docker run -itd --name harshc1q1 alpine:harsh
  207  ls
  208  vim Dockerfile
  209  docker build -t alpine:harsh .
  210  docker run -itd --name harshc1q1 alpine:harsh
  211  docker rm harshc1q1
  212  docker run -itd --name harshc1q1 alpine:harsh
  213  ls
  214  cd
  215  ls
  216  cd question1/
  217  ls
  218  cd /var/lib/docker
  219  ;s

history/.mozilla/firefox/za1pqe9a.default-release/prefs.js
// Mozilla User Preferences

// DO NOT EDIT THIS FILE.
//
// If you make changes to this file while the application is running,
// the changes will be overwritten when the application exits.
//
// To change a preference value, you can either:
// - modify it via the UI (e.g. via about:config in the browser); or
// - set it within a user.js file in your profile.

user_pref("app.normandy.first_run", false);
user_pref("app.normandy.migrationsApplied", 10);
user_pref("app.normandy.startupExperimentPrefs.browser.newtabpage.activity-stream.discoverystream.region-layout-config", "US,CA,GB");
user_pref("app.normandy.startupExperimentPrefs.browser.newtabpage.activity-stream.discoverystream.region-stories-config", "US,DE,CA,GB");
user_pref("app.normandy.startupRolloutPrefs.app.normandy.onsync_skew_sec", 3300);
user_pref("app.normandy.user_id", "3749b190-37e3-4eb2-a3fd-a41c179e14c2");
user_pref("app.update.lastUpdateTime.addon-background-update-timer", 1595992587);
user_pref("app.update.lastUpdateTime.blocklist-background-update-timer", 1595904113);
user_pref("app.update.lastUpdateTime.browser-cleanup-thumbnails", 1596046825);
user_pref("app.update.lastUpdateTime.recipe-client-addon-run", 1596046606);
user_pref("app.update.lastUpdateTime.rs-experiment-loader-timer", 1596044474);
user_pref("app.update.lastUpdateTime.search-engine-update-timer", 1596047024);
user_pref("app.update.lastUpdateTime.services-settings-poll-changes", 1595992467);
user_pref("app.update.lastUpdateTime.telemetry_modules_ping", 1595903636);
user_pref("app.update.lastUpdateTime.xpi-signature-verification", 1596024771);
user_pref("browser.bookmarks.editDialog.confirmationHintShowCount", 1);
user_pref("browser.bookmarks.restore_default_bookmarks", false);
user_pref("browser.bookmarks.showMobileBookmarks", false);
user_pref("browser.cache.disk.amount_written", 1085438);
user_pref("browser.cache.disk.capacity", 1048576);
user_pref("browser.cache.disk.filesystem_reported", 1);
user_pref("browser.contentblocking.category", "standard");
user_pref("browser.download.panel.shown", true);
user_pref("browser.laterrun.bookkeeping.profileCreationTime", 1595898962);
user_pref("browser.laterrun.bookkeeping.sessionCount", 22);
user_pref("browser.laterrun.enabled", true);
user_pref("browser.migration.version", 96);
user_pref("browser.newtabpage.activity-stream.impressionId", "{e6c7ee25-109f-45c7-99dd-ec62ef6f18a7}");
user_pref("browser.newtabpage.pinned", "[null,null,{\"url\":\"https://www.canva.com/\"},{\"url\":\"https://www.hackerrank.com/dashboard\",\"label\":\"Hackerrank\",\"baseDomain\":\"hackerrank.com\"},null,{\"url\":\"https://github.com/\",\"label\":\"github\",\"baseDomain\":\"github.com\"},{\"url\":\"https://mail.google.com/mail/u/2/#inbox\",\"label\":\"Mail\",\"baseDomain\":\"mail.google.com\"},{\"url\":\"https://www.hackerearth.com/\",\"baseDomain\":\"hackerearth.com\"}]");
user_pref("browser.newtabpage.storageVersion", 1);
user_pref("browser.pageActions.persistedActions", "{\"version\":1,\"ids\":[\"bookmark\",\"pinTab\",\"bookmarkSeparator\",\"copyURL\",\"emailLink\",\"addSearchEngine\",\"sendToDevice\",\"pocket\",\"screenshots_mozilla_org\"],\"idsInUrlbar\":[\"pocket\",\"bookmark\"]}");
user_pref("browser.pagethumbnails.storage_version", 3);
user_pref("browser.rights.3.shown", true);
user_pref("browser.safebrowsing.provider.google4.lastupdatetime", "1596048583255");
user_pref("browser.safebrowsing.provider.google4.nextupdatetime", "1596050359255");
user_pref("browser.safebrowsing.provider.mozilla.lastupdatetime", "1596046808968");
user_pref("browser.safebrowsing.provider.mozilla.nextupdatetime", "1596050408968");
user_pref("browser.search.region", "IN");
user_pref("browser.sessionstore.upgradeBackup.latestBuildID", "20200709140757");
user_pref("browser.shell.mostRecentDateSetAsDefault", "1596044470");
user_pref("browser.slowStartup.averageTime", 9206);
user_pref("browser.slowStartup.samples", 2);
user_pref("browser.startup.homepage_override.buildID", "20200709140757");
user_pref("browser.startup.homepage_override.mstone", "78.0.2");
user_pref("browser.startup.lastColdStartupCheck", 1596044465);
user_pref("browser.uiCustomization.state", "{\"placements\":{\"widget-overflow-fixed-list\":[],\"nav-bar\":[\"back-button\",\"forward-button\",\"stop-reload-button\",\"home-button\",\"customizableui-special-spring1\",\"urlbar-container\",\"customizableui-special-spring2\",\"downloads-button\",\"library-button\",\"sidebar-button\",\"fxa-toolbar-menu-button\",\"_7a7a4a92-a2a0-41d1-9fd7-1e92480d612d_-browser-action\",\"chrome-gnome-shell_gnome_org-browser-action\",\"_6ac85730-7d0f-4de0-b3fa-21142dd85326_-browser-action\"],\"toolbar-menubar\":[\"menubar-items\"],\"TabsToolbar\":[\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\"],\"PersonalToolbar\":[\"personal-bookmarks\"]},\"seen\":[\"developer-button\",\"_7a7a4a92-a2a0-41d1-9fd7-1e92480d612d_-browser-action\",\"chrome-gnome-shell_gnome_org-browser-action\",\"_6ac85730-7d0f-4de0-b3fa-21142dd85326_-browser-action\"],\"dirtyAreaCache\":[\"nav-bar\",\"toolbar-menubar\",\"TabsToolbar\",\"PersonalToolbar\"],\"currentVersion\":16,\"newElementCount\":2}");
user_pref("browser.urlbar.placeholderName", "Google");
user_pref("browser.urlbar.placeholderName.private", "Google");
user_pref("browser.urlbar.tipShownCount.searchTip_onboard", 4);
user_pref("datareporting.policy.dataSubmissionPolicyAcceptedVersion", 2);
user_pref("datareporting.policy.dataSubmissionPolicyNotifiedTime", "1595898968890");
user_pref("distribution.fedora.bookmarksProcessed", true);
user_pref("distribution.iniFile.exists.appversion", "78.0.2");
user_pref("distribution.iniFile.exists.value", true);
user_pref("doh-rollout.balrog-migration-done", true);
user_pref("dom.push.userAgentID", "979120e65164457aa72a25f2dd70b170");
user_pref("extensions.activeThemeID", "default-theme@mozilla.org");
user_pref("extensions.blocklist.pingCountVersion", -1);
user_pref("extensions.databaseSchema", 32);
user_pref("extensions.getAddons.cache.lastUpdate", 1595992590);
user_pref("extensions.getAddons.databaseSchema", 6);
user_pref("extensions.incognito.migrated", true);
user_pref("extensions.lastAppBuildId", "20200709140757");
user_pref("extensions.lastAppVersion", "78.0.2");
user_pref("extensions.lastPlatformVersion", "78.0.2");
user_pref("extensions.pendingOperations", false);
user_pref("extensions.systemAddonSet", "{\"schema\":1,\"addons\":{}}");
user_pref("extensions.ui.dictionary.hidden", true);
user_pref("extensions.ui.lastCategory", "addons://list/extension");
user_pref("extensions.ui.locale.hidden", true);
user_pref("extensions.webcompat.perform_injections", true);
user_pref("extensions.webcompat.perform_ua_overrides", true);
user_pref("extensions.webextensions.ExtensionStorageIDB.migrated.chrome-gnome-shell@gnome.org", true);
user_pref("extensions.webextensions.ExtensionStorageIDB.migrated.doh-rollout@mozilla.org", true);
user_pref("extensions.webextensions.ExtensionStorageIDB.migrated.screenshots@mozilla.org", true);
user_pref("extensions.webextensions.ExtensionStorageIDB.migrated.{6AC85730-7D0F-4de0-B3FA-21142DD85326}", true);
user_pref("extensions.webextensions.ExtensionStorageIDB.migrated.{7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}", true);
user_pref("extensions.webextensions.uuids", "{\"doh-rollout@mozilla.org\":\"47255336-c4a4-4348-bcf3-61a789f88a7e\",\"formautofill@mozilla.org\":\"7780a5d9-9454-4840-a6fc-efc84a1c28f1\",\"screenshots@mozilla.org\":\"02c52b77-691b-4334-8f63-4ce279faf3da\",\"webcompat-reporter@mozilla.org\":\"307ec7e4-6592-4dd4-a562-462b720ff89e\",\"webcompat@mozilla.org\":\"25e551bb-4c2d-4ab8-868f-1f8b8773772a\",\"default-theme@mozilla.org\":\"2cd75cf2-fb6a-4bd3-896e-f2ba5b3ea9e6\",\"google@search.mozilla.org\":\"3a4c3e6e-6832-429f-aea9-b88849343418\",\"amazondotcom@search.mozilla.org\":\"e3373dd7-96ba-4b71-8e56-82e1ff98435a\",\"bing@search.mozilla.org\":\"4e294dcc-4fcb-4650-83da-7e4d155f3a64\",\"ddg@search.mozilla.org\":\"a6151d8f-3eac-45da-88dd-3804e0d0e53a\",\"ebay@search.mozilla.org\":\"7c0579b9-7802-4fe0-86c1-897e20bf8d2e\",\"twitter@search.mozilla.org\":\"9e6b1990-8eab-4514-962e-79a215ce4641\",\"wikipedia@search.mozilla.org\":\"6e140c2a-d995-477e-a748-1028cc11a18b\",\"{7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}\":\"7452ce82-6be5-4193-8e0c-5eb5b2f488fd\",\"chrome-gnome-shell@gnome.org\":\"0034c6b3-d4ff-4e3a-a464-4b1eb48891fb\",\"{6AC85730-7D0F-4de0-B3FA-21142DD85326}\":\"6f9226a1-3bee-40b4-b7e0-81ee55cb4088\"}");
user_pref("identity.fxaccounts.account.device.name", "harsh’s Firefox on localhost");
user_pref("identity.fxaccounts.account.telemetry.sanitized_uid", "5ada4b1683ff2a270a2a1d3680dfed64");
user_pref("identity.fxaccounts.commands.missed.last_fetch", 1595992313);
user_pref("identity.fxaccounts.lastSignedInUserHash", "mul822QHgCeL4e3d+wVCCwoe4CVndle0EQyJ7k4IRF0=");
user_pref("identity.fxaccounts.toolbar.accessed", true);
user_pref("idle.lastDailyNotification", 1596033419);
user_pref("media.gmp-manager.buildID", "20200709140757");
user_pref("media.gmp-manager.lastCheck", 1596030121);
user_pref("media.gmp.storage.version.observed", 1);
user_pref("network.predictor.cleaned-up", true);
user_pref("pdfjs.enabledCache.state", true);
user_pref("pdfjs.migrationVersion", 2);
user_pref("places.database.lastMaintenance", 1595903719);
user_pref("places.history.expiration.transient_current_max_pages", 104858);
user_pref("pref.privacy.disable_button.cookie_exceptions", false);
user_pref("privacy.purge_trackers.date_in_cookie_database", "0");
user_pref("privacy.sanitize.pending", "[{\"id\":\"newtab-container\",\"itemsToClear\":[],\"options\":{}}]");
user_pref("security.remote_settings.crlite_filters.checked", 1595992467);
user_pref("security.remote_settings.intermediates.checked", 1595992467);
user_pref("security.sandbox.content.tempDirSuffix", "5d6d034d-edca-40da-b85c-019a3e38941a");
user_pref("security.sandbox.plugin.tempDirSuffix", "f9baf9e4-7989-433f-919c-844188a904c6");
user_pref("services.blocklist.addons.checked", 1595992467);
user_pref("services.blocklist.gfx.checked", 1595992467);
user_pref("services.blocklist.pinning.checked", 1595992467);
user_pref("services.blocklist.plugins.checked", 1595992467);
user_pref("services.settings.clock_skew_seconds", 0);
user_pref("services.settings.last_etag", "\"1595980879718\"");
user_pref("services.settings.last_update_seconds", 1595992467);
user_pref("services.settings.main.anti-tracking-url-decoration.last_check", 1595992467);
user_pref("services.settings.main.cfr-fxa.last_check", 1595992467);
user_pref("services.settings.main.cfr.last_check", 1595992467);
user_pref("services.settings.main.fxmonitor-breaches.last_check", 1595992467);
user_pref("services.settings.main.hijack-blocklists.last_check", 1595992467);
user_pref("services.settings.main.language-dictionaries.last_check", 1595992467);
user_pref("services.settings.main.message-groups.last_check", 1595992467);
user_pref("services.settings.main.messaging-experiments.last_check", 1595992467);
user_pref("services.settings.main.normandy-recipes-capabilities.last_check", 1595992467);
user_pref("services.settings.main.public-suffix-list.last_check", 1595992467);
user_pref("services.settings.main.search-config.last_check", 1595992467);
user_pref("services.settings.main.search-default-override-allowlist.last_check", 1595992467);
user_pref("services.settings.main.sites-classification.last_check", 1595992467);
user_pref("services.settings.main.tippytop.last_check", 1595992467);
user_pref("services.settings.main.url-classifier-skip-urls.last_check", 1595992467);
user_pref("services.settings.main.whats-new-panel.last_check", 1595992467);
user_pref("services.settings.security.onecrl.checked", 1595992467);
user_pref("services.sync.addons.lastSync", "1595903808.29");
user_pref("services.sync.addons.syncID", "pon_opSx0nUT");
user_pref("services.sync.bookmarks.lastSync", "1596030155.71");
user_pref("services.sync.bookmarks.syncID", "4JKFYVfYUhNj");
user_pref("services.sync.client.GUID", "fxR2xnFPYxJR");
user_pref("services.sync.client.syncID", "L9aBiP2A2RBj");
user_pref("services.sync.clients.devices.desktop", 3);
user_pref("services.sync.clients.devices.mobile", 0);
user_pref("services.sync.clients.lastRecordUpload", 1596044487);
user_pref("services.sync.clients.lastSync", "1596044487.63");
user_pref("services.sync.clients.syncID", "09FdTqNgiCAM");
user_pref("services.sync.declinedEngines", "");
user_pref("services.sync.deletePwdFxA", true);
user_pref("services.sync.doctor.lastRepairAdvance", 1595992001);
user_pref("services.sync.engine.prefs.modified", false);
user_pref("services.sync.extension-storage.lastSync", "0");
user_pref("services.sync.forms.lastSync", "1596049420.62");
user_pref("services.sync.forms.syncID", "5PN3J83A67c5");
user_pref("services.sync.globalScore", 7);
user_pref("services.sync.history.lastSync", "1596049421.28");
user_pref("services.sync.history.syncID", "WA1NYXRoc6cZ");
user_pref("services.sync.lastPing", 1595991997);
user_pref("services.sync.lastSync", "Thu Jul 30 2020 00:33:41 GMT+0530 (India Standard Time)");
user_pref("services.sync.lastTabFetch", 1596049420);
user_pref("services.sync.nextSync", 1596050021);
user_pref("services.sync.passwords.lastSync", "1596031516.86");
user_pref("services.sync.passwords.syncID", "aKzfP_97MSA1");
user_pref("services.sync.prefs.lastSync", "1595903712.68");
user_pref("services.sync.prefs.syncID", "ESKlc-Z1NQzd");
user_pref("services.sync.syncInterval", 3600000);
user_pref("services.sync.syncThreshold", 300);
user_pref("services.sync.tabs.lastSync", "1596049420.13");
user_pref("services.sync.tabs.syncID", "_nzM6fTd_NYh");
user_pref("services.sync.username", "harsh.mangal1108@gmail.com");
user_pref("signon.importedFromSqlite", true);
user_pref("storage.vacuum.last.index", 1);
user_pref("storage.vacuum.last.places.sqlite", 1595940538);
user_pref("toolkit.startup.last_success", 1596044458);
user_pref("toolkit.telemetry.cachedClientID", "6fc5627b-46c1-4cab-8407-5c6924aed473");
user_pref("toolkit.telemetry.previousBuildID", "20200709140757");
user_pref("toolkit.telemetry.reportingpolicy.firstRun", false);
user_pref("trailhead.firstrun.didSeeAboutWelcome", true);
